### ISSUE TITLE
Feature Request: Check Pacman lock and wait for already running pacman operations

### ISSUE THREAD

**Issue by RubenKelevra** [fosskers/aura#918]
If paru is tasked with installation of an Aur package, and the package building is completed, paru will check the lock of pacman to determine if there's already a running pacman instance. If so, paru will wait until the lock file is removed and then continue.

Aura on the other hand will just fail hard if there's an already running pacman instance:

```
$ aura -A firefox-nightly-de-bin --noconfirm
[...]
==> Finished making: firefox-nightly-de-bin 131.0a1.20240821-1 (Wed 21 Aug 2024 04:10:03 PM CEST)
[sudo] password for ruben: 
error: failed to init transaction (unable to lock database)
error: could not lock database: File exists
  if you're sure a package manager is not already
  running, you can remove /var/lib/pacman/db.lck
aura :: A call to pacman -U failed.

```

It would be nice if Aura could do the same, and check for the lock file. If it's removed, wait a couple of seconds randomly, to avoid race conditions with other instances of Aura also waiting, or another `pacman` call issued by another script, and check again. If still gone, continue.

**Comment by fosskers** [comment]
Reasonable, thanks.

**Comment by RubenKelevra** [comment]
Thanks!

**Comment by RubenKelevra** [comment]
Thanks! One minute however is a bit short.

How about an hour or so?

Usually programs will compile / install in less than an hour. If it takes longer it's either a massive program or pacman crashed somehow uncleanly.

**Comment by fosskers** [comment]
The time that Pacman is holding the lock is/should be relatively short. Aura can meanwhile build packages, and it's only right before the install attempt that it'll check if the lock is there. If that still fails, you can always `-C` to retrieve the version you just built.

**Comment by RubenKelevra** [comment]
Well that depends on how fast your download speed is.

In the situation I've noticed it was: I was running a system update while the aura was downloading the binary of Firefox.

While system updates are often finished after a couple of seconds, this time I needed to download more than 2 GBs and my mobile provider was slow, so it took 10-15 minutes to complete.

As I said, 1 hour would be a good threshold to avoid getting stuck forever.

Another option would be to check for a running pacman process and fail early if there is none.
